deposit-transactions.md

```diff
@@ -0,0 +1,165 @@
+
+[l1-attributes-depositor-account]: #l1-attributes-depositor-account
+
+The Depositor Account is an EOA with no known private key. It has the address
+`0xdeaddeaddeaddeaddeaddeaddeaddeaddead0001`. Its value returned by the `CALLER` and `ORIGIN`
+opcodes during execution of the L1 Attributes Deposit Transaction.
+
+### L1 Attributes Predeploy
+
+[l1-attributes-predeploy]: #l1-attributes-predeploy
+
+A predeployed contract on L2 at address `0x4200000000000000000000000000000000000014`, which holds
+certain block variables from the corresponding L1 block in storage, so that they may be accessed
+during the execution of the subsequent deposited transactions.
+
+The contract implements an authorization scheme, such that it only accepts state-changing calls from
+the [Depositor Account].
+
+The contract has the following solidity interface, and can be interacted with according to the
+[contract ABI specification][ABI].
+
+[ABI]: https://docs.soliditylang.org/en/v0.8.10/abi-spec.html
+
+```solidity
+interface L1BlockValues {
+
+  function setL1BlockValues(
+    uint256 number,
+    uint256 timestamp,
+    uint256 baseFee,
+    bytes32 hash
+  ) external;
+
+  function l1Number() view;
+  function l1Timestamp() view;
+  function l1BaseFee() view;
+  function l1Hash() view;
+}
+```
+
+## Deposited Transactions
+
+[deposited-transactions]: #deposited-transactions
+
+"Deposited Transactions" are derived from logs emitted by the [Deposit Feed
+contract][deposit-feed-contract] on L1.
+
+### Deposit Feed Contract
+
+[deposit-feed-contract]: #deposit-feed-contract
+
+The Deposit Feed contract is deployed to L1. Deposited Transactions are derived from the values in
+the `TransactionDeposited` event(s) emitted by the Deposit Feed contract.
+
+The Deposit Feed handles two special cases:
+
+1. A `to` value of `0`, which results in a contract creation on L2.
+2. A call from a contract account, in which case the `from` value is transformed to its L2 alias.
+   This prevents attacks in which a contract on L1 has the same address as a contract on L2 but
+   doesn't have the same code. We can safely ignore this for EOAs because they're guaranteed to have
+   the same "code" (i.e. no code at all). This also makes it possible for users to interact with
+   contracts on L2 even when the Sequencer is down.
+
+A solidity like pseudocode implementation demonstrates the functionality:
+
+```solidity
+contract DepositFeed {
+  event TransactionDeposited(
+    address indexed from,
+    address indexed to,
+    uint256 value,
+    bytes _data
+  );
+
+  event TransactionDeposited(
+    address indexed from,
+    uint256 value,
+    bytes _data
+  );
+
+  function depositTransaction(
+    address to,
+    uint256 value,
+    bytes memory _data
+  ) external {
+    address from;
+    if (msg.sender == tx.origin) {
+        from = msg.sender;
+    } else {
+        from = msg.sender + 0x1111000000000000000000000000000000001111;
+    }
+
+    if(_to == address(0)) {
```

Comment:
I think using addr(0) is *probably* fine, but raising [this](https://dapp.org.uk/reports/optimism.html#org9346693) issue we've seen in the past. I don't think it's the same issue, but there's always risk of confusion where a dev allows an L1 contract to call an arbitrary L2 contract, and a user can trick the contract into CREATEing instead of making a call to `addr(0)`

Reply:
Good point to consider. Going to resolve this comment and refer to it in the thread above which is closely related.